### PR TITLE
lfe: 1.2.1 -> 1.3

### PR DESCRIPTION
**Pull request by yurrriq**
###### Motivation for this change

Hack together an `lfe_1_3` derivation and make it the default `lfe`.

Notably, the following now works.

```lfe
(include-lib "lfe/include/clj.lfe")
(-> 2 (+ 2) (== 4)) ; true
```

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by xeji** [comment]
Why do we need to keep the old version?

**Comment by yurrriq** [comment]
We keep old versions of other BEAM languages because many people still use them. In the case of LFE, there aren't very many users, especially on Nix, so it's more for uniformity across BEAM packages  than anything else.

File: pkgs/development/beam-modules/default.nix [hunk 1]
```diff
@@ -64,8 +64,9 @@ let
                        debugInfo = true;
                      };
 
-        lfe = lfe_1_2;
+        lfe = lfe_1_3;
         lfe_1_2 = lib.callLFE ../interpreters/lfe/1.2.nix { inherit erlang buildRebar3 buildHex; };
+        lfe_1_3 = lib.callLFE ../interpreters/lfe/1.3.nix { inherit erlang buildRebar3 buildHex; };
```
Comment (Mic92) [inline pkgs/development/beam-modules/default.nix]: Do we need both lfe_1_2 and 1.3? The rule of thumb is, that we usually do this for projects where multiple maintained major versions exists. This seems not the case here.

Comment (xeji) [reply pkgs/development/beam-modules/default.nix]: see Q+A above :smiley:

**Comment by Mic92** [comment]
@yurrriq I suppose the old erlang platforms are still maintained upstream and only major versions are kept. I am not familiar with LFE, will older versions still receive updates? Where there breaking changes in the last minor release? https://github.com/rvirding/lfe/compare/v1.2.1...v1.3

**Comment by yurrriq** [comment]
I really don't use LFE anymore, so I'm not motivated to get this working.

**Pull request closed without merging**